ESMF/GOCART2G_GridComp/DU2G_GridComp/DU2G_GridCompMod.F90

```diff
@@ -516,50 +518,41 @@ subroutine Initialize (GC, import, export, clock, RC)
 
 !   Create Radiation Mie Table
 !   --------------------------
-    call MAPL_GetResource (MAPL, NUM_BANDS, 'NUM_BANDS:', __RC__)
 
 !   Get file names for the optical tables
-    call ESMF_ConfigGetAttribute (cfg, self%rad_MieTable(instance)%optics_file, &
+    call ESMF_ConfigGetAttribute (cfg, file_, &
                                   label="aerosol_radBands_optics_file:", __RC__ )
 
-    allocate (self%rad_MieTable(instance)%channels(NUM_BANDS), __STAT__ )
-    self%rad_MieTable(instance)%nch = NUM_BANDS
-
     call ESMF_ConfigFindLabel(cfg, label="BANDS:", isPresent=bands_are_present, __RC__)
 
     if (bands_are_present) then
-       call ESMF_ConfigGetAttribute (cfg, self%rad_MieTable(instance)%channels, label= "BANDS:", &
-                                    count=self%rad_MieTable(instance)%nch, __RC__)
+       NUM_BANDS = ESMF_ConfigGetLen(cfg, label= "BANDS:", __RC__)
+       allocate (channels_(NUM_BANDS), __STAT__ )
+       call ESMF_ConfigGetAttribute (cfg, channels_, count=NUM_BANDS, label= "BANDS:", __RC__)
     else
+       call MAPL_GetResource (MAPL, NUM_BANDS, 'NUM_BANDS:', __RC__)
+       allocate (channels_(NUM_BANDS), __STAT__ )
        do i = 1, NUM_BANDS
-          self%rad_MieTable(instance)%channels(i) = i
+          channels_(i) = i
        end do
     endif
-
-    allocate (self%rad_MieTable(instance)%mie_aerosol, __STAT__)
-    self%rad_MieTable(instance)%mie_aerosol = Chem_MieTableCreate (self%rad_MieTable(instance)%optics_file, __RC__)
-    call Chem_MieTableRead (self%rad_MieTable(instance)%mie_aerosol, NUM_BANDS, self%rad_MieTable(instance)%channels, __RC__)
-
+    self%rad_Mie(instance) = Chem_Mie2G(file_, channels_*1.e-9, __RC__)
+ 
```

Comment:
Something is still wrong in this section.   The argument to `Chem_Mie2G` looks like a wavelength, but channels is an integer.  I think the variable should be renamed to `wavelengths`.   The loop for line 536 still looks odd either way, and needs attention from@amdasilva.

Reply:
channels becomes wavelength after multiplying by 1.e-9

Reply:
I think we will still use 'channels"  here because they are integers reading from RC file

Reply:
I have no idea what the loop ia about. https://github.com/GEOS-ESM/GOCART/blob/4f2bdc182239186fc1353b3048f65b233cfeb7a0/ESMF/GOCART2G_GridComp/DU2G_GridComp/DU2G_GridCompMod.F90#L535-L537